### PR TITLE
Add tracing support

### PR DESCRIPTION
**Pull request by jamieklassen**
Add tracing support for zipkin/opentracing

### CODE REVIEW

**Comment by ericbottard** [comment]
I'll have a closer look at this when I get back from vacation, but it seems to me that switching dependency management tooling to work around some build issues is a real code smell. Surely there is a better way. If you could try to fix it with `dep` that'd be cool, else please write down enough context so that I can try to fix this myself.

TTY on Monday

File: pkg/dispatcher/http/http.go [hunk 1]
```diff
@@ -51,6 +64,6 @@ func (httpDispatcher) Dispatch(in interface{}) (interface{}, error) {
 	return string(out), nil
 }
 
-func NewHttpDispatcher() dispatcher.Dispatcher {
-	return httpDispatcher{""}
+func NewHttpDispatcher(traceContext dispatcher.TraceContext) dispatcher.Dispatcher {
+	return &httpDispatcher{traceContext: traceContext}
```
Comment (ericbottard) [inline pkg/dispatcher/http/http.go]: Why the change to pointer here?

**Comment by ericbottard** [comment]
Looked at this a bit further. First question that comes to mind is: why would the tracing be _per-Dispatcher_

We won't control (always?) the other side of the invocation (ie won't be able to talk to zipkin on the `main` container)

**Comment by markfisher** [comment]
closing for now, we need to revisit tracing and can refer to this in the archive as needed
